pandas/core/frame.py

```diff
@@ -10719,16 +10725,28 @@ def map(
         0   1.000000   4.494400
         1  11.262736  20.857489
         """
-        if na_action not in {"ignore", None}:
-            raise ValueError(f"na_action must be 'ignore' or None. Got {na_action!r}")
+        if na_action != lib.no_default:
```

Comment:
In case it's not obvious, I'm using `lib.no_default` as a sentinel to know if `na_action` was set by the user, so we can show them a warning. If I leave the default to `None` I can't tell if the user call was `.map(func)` (no warning needed) or `.map(func, na_action=None)` (warning needed)